### PR TITLE
(#927) What is the expected behaviour of BehavesAsMap?

### PR DESCRIPTION
**Pull request by Umbrah**
This is for #927 

Updated matchers inside BehavesAsMap.matchesSafely().
Updated tests which uses BehavesAsMap.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1099](https://github.com/yegor256/cactoos/issues/1099) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1099](https://github.com/yegor256/cactoos/issues/1099) is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C63314D6Z/15f61a6f-dad0-4675-95c9-8ee299ddb534); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by victornoel** [comment]
@Umbrah about the updated tests: I think it would be safer to leave the multiple entries and modify the use of `BehavesAsMap` so that it check one of the entries.

@Umbrah @llorllale now the `BehavesAsMap` `Matcher` is a wrapper around `IsMapContaining`: do one of you have any ideas of extra checks that could be included in this class to improve its behaviour? We could add a `@todo` to implement it later at least.

**Comment by Umbrah** [comment]
@victornoel I updated tests according to your suggestion. No ideas from my side about extra checks here, map logic is pretty simple - not much to check :)

**Comment by victornoel** [comment]
@Umbrah thanks

@llorllale it's good to merge, except maybe if you think we should add a todo as discussed here: https://github.com/yegor256/cactoos/pull/1099#issuecomment-480592187

**Comment by victornoel** [comment]
@llorllale ping

File: src/test/java/org/cactoos/map/BehavesAsMap.java [hunk 1]
```diff
@@ -69,27 +67,9 @@ public boolean matchesSafely(final Map<K, V> map) {
...
-            map,
-            new IsMapContaining<>(
-                new IsAnything<>(),
                 new IsEqual<>(this.value)
             )
         );
-        MatcherAssert.assertThat(
-            "Doesn't contain the key in #keySet()",
-            map.keySet(),
-            new IsCollectionContaining<>(new IsEqual<>(this.key))
```
Comment (llorllale) [inline src/test/java/org/cactoos/map/BehavesAsMap.java]: @Umbrah why remove this check?

File: src/test/java/org/cactoos/map/BehavesAsMap.java [hunk 2]
```diff
@@ -69,27 +67,9 @@ public boolean matchesSafely(final Map<K, V> map) {
...
-        );
-        MatcherAssert.assertThat(
-            "Doesn't contain the value in #values()",
-            map.values(),
-            new IsCollectionContaining<>(new IsEqual<>(this.value))
```
Comment (llorllale) [inline src/test/java/org/cactoos/map/BehavesAsMap.java]: @Umbrah why remove this check?

File: src/test/java/org/cactoos/map/MapOfTest.java [hunk 3]
```diff
@@ -59,7 +59,7 @@ public void behavesAsMap() {
                     new MapEntry<>(1, 1)
                 )
             ),
-            new BehavesAsMap<>(0, 1)
+            new BehavesAsMap<>(1, 1)
```
Comment (llorllale) [inline src/test/java/org/cactoos/map/MapOfTest.java]: @Umbrah why this change?

File: src/test/java/org/cactoos/map/SolidTest.java [hunk 4]
```diff
@@ -55,7 +55,7 @@ public void behavesAsMap() {
                 new MapEntry<>(0, -1),
                 new MapEntry<>(1, 1)
             ),
-            new BehavesAsMap<>(0, 1)
+            new BehavesAsMap<>(1, 1)
```
Comment (llorllale) [inline src/test/java/org/cactoos/map/SolidTest.java]: @Umbrah why this change?

**Comment by llorllale** [comment]
@Umbrah left a couple of questions

File: src/test/java/org/cactoos/map/BehavesAsMap.java (hunk 1 above)
Comment (Umbrah) [reply src/test/java/org/cactoos/map/BehavesAsMap.java]: I wanted this class to check for a key-value pair in a map but not to check for it separately(as it was previously). Do you think we should leave checking for a key in key-set?

File: src/test/java/org/cactoos/map/BehavesAsMap.java (hunk 2 above)
Comment (Umbrah) [reply src/test/java/org/cactoos/map/BehavesAsMap.java]: I wanted this class to check for a key-value pair in a map but not to check for it separately(as it was previously). Do you think we should leave checking for a value in values?

File: src/test/java/org/cactoos/map/BehavesAsMap.java (hunk 1 above)
Comment (Umbrah) [reply src/test/java/org/cactoos/map/BehavesAsMap.java]: Hm, now when we have checked that map contains a key-value pair(but not a separate key and separate value), verifying keySet and values looks good for me, I think I can return these checks back. Please let me know what do you think

**Comment by Umbrah** [comment]
@llorllale I returned back checks for keySet and values, please check

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
Code review was too long (10 days), architects (@llorllale) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1099](https://github.com/yegor256/cactoos/issues/1099) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)
